docs/markdown/Python/python/python-lockfiles.md

```diff
@@ -0,0 +1,271 @@
+---
+title: "Lockfiles"
+slug: "python-lockfiles"
+excerpt: "Securely locking down your third-party dependencies."
+hidden: false
+createdAt: "2023-03-11T01:50:46.369Z"
+updatedAt: "2023-03-11T01:50:46.369Z"
+---
+Third-party dependencies are typically specified via a range of allowed versions, known as "requirements", in a file such as requirements.txt or pyproject.toml. A dependency resolution tool like Pip or Poetry then takes these initial requirements and attempts to find and download a consistent set of transitive dependencies that are mutually compatible with each other and with the target Python interpreter version.
```

Comment:
Words get in the way here. The word range seems like it more strongly associates with something like `>=1.2.3` (for 1 project as a specifier) than it does a list of (range of) requirement strings.

Reply:
Ah yes, I intended the former (a range of versions), I'll rephrase.